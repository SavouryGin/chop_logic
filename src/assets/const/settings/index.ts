import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
};
