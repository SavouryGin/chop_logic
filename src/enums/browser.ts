export enum Browser {
  Unknown,
  Opera,
  Chrome,
  Safari,
  Firefox,
}
