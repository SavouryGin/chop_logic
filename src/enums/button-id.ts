export enum ButtonID {
  Cancel,
  Navigation,
  ColorTheme,
  Sounds,
  Settings,
  FullScreen,
  Tools,
  Apply,
  Premise,
  Reiteration,
  Replace,
  Delete,
  ImplicationCreation,
  ImplicationDistribution,
  ContradictionRealization,
  ImplicationElimination,
  Assumption,
  NotIntroduction,
  NotElimination,
  AndIntroduction,
  AndElimination,
  OrIntroduction,
  OrElimination,
  ImpliesIntroduction,
  ImpliesElimination,
  EquivIntroduction,
  EquivElimination,
  Shortcut,
  SavePDF,
  ExportXML,
  ImportXML,
  CopyProof,
  PasteProof,
  CutProof,
  Reset,
}
