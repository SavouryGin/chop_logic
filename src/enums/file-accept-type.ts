export enum FileAcceptType {
  Image = 'image/*',
  Video = 'video/*',
  Audio = 'audio/*',
  XML = 'text/xml',
  PDF = 'application/pdf',
  CSV = '.csv',
  Text = 'text/plain',
  Excel = 'application/vnd.ms-excel, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
}
