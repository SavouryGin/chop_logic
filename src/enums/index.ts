export enum Icon {
  Default = 'chop-icon__check',
  Home = 'chop-icon__home',
  Menu = 'chop-icon__menu',
  Enlarge = 'chop-icon__enlarge2',
  Shrink = 'chop-icon__shrink2',
  Settings = 'chop-icon__cog',
  Delete = 'chop-icon__bin2',
  Up = 'chop-icon__circle-up',
  Down = 'chop-icon__circle-down',
  Right = 'chop-icon__circle-right',
  Left = 'chop-icon__circle-left',
  Sound = 'chop-icon__volume-medium',
  NoSound = 'chop-icon__volume-mute2',
  LightMode = 'chop-icon__sun',
  DarkMode = 'chop-icon__contrast',
}

export enum Pages {
  Home,
  NotFound,
  Propositions,
  Predicates,
  TruthTables,
  Syllogisms,
}
