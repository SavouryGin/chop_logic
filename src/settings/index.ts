import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const languageStringOptions = ['en', 'ru'];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  linkedIn: 'https://www.linkedin.com/in/dmitrii-suroviagin/',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
  facebook: 'https://www.facebook.com/dmitrii.suroviagin',
};
