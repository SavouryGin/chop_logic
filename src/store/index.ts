import createSagaMiddleware from 'redux-saga';
import rootReducer from './root-reducer';
import rootSaga from './root-saga';
import { configureStore } from '@reduxjs/toolkit';

const sagaMiddleware = createSagaMiddleware();

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat([sagaMiddleware]),
});

sagaMiddleware.run(rootSaga);
