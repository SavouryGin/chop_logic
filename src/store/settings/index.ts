import { Language } from 'types';
import { PayloadAction, createSlice } from '@reduxjs/toolkit';
import { SettingsFlag, SettingsInitialState } from './interfaces';

export const settingsInitialState: SettingsInitialState = {
  language: 'en',
  flags: {
    isNavigationOpen: false,
    isSidebarOpen: false,
    isDarkMode: false,
    isFullScreen: false,
    isSettingOpen: false,
    isSoundsEnabled: false,
  },
};

export const settingsSlice = createSlice({
  name: 'settings',
  initialState: settingsInitialState,
  reducers: {
    resetSettings: () => {
      return settingsInitialState;
    },

    toggleFlag: (state, action: PayloadAction<SettingsFlag>) => {
      const currentFlagValue = state.flags[action.payload];
      state.flags[action.payload] = !currentFlagValue;
    },

    setLanguage: (state, action: PayloadAction<Language>) => {
      state.language = action.payload;
    },

    setFullScreenFlag: (state, action: PayloadAction<boolean>) => {
      state.flags.isFullScreen = action.payload;
    },

    setUpFlag: (state, action: PayloadAction<{ flag: SettingsFlag; value: boolean }>) => {
      const { flag, value } = action.payload;
      state.flags[flag] = value;
    },
  },
});

export const settingsActions = { ...settingsSlice.actions };
