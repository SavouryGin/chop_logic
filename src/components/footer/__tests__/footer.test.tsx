import Footer from '../index';
import React from 'react';
import renderWithRedux from 'utils/testing/render-with-redux';
import { Icon } from 'enums';
import { combineReducers } from '@reduxjs/toolkit';
import { externalLinks } from 'utils/settings';
import { screen } from '@testing-library/react';
import { settingsInitialState, settingsSlice } from 'store/settings';

describe('Footer component:', () => {
  const testProps = {
    className: 'test-class',
    isDarkMode: false,
  };

  const mockedReducer = combineReducers({ settings: settingsSlice.reducer });
  const mockedState = { settings: settingsInitialState };

  beforeEach(() => {
    renderWithRedux(<Footer {...testProps} />, mockedReducer, mockedState);
  });

  it('renders the footer container', () => {
    const footer = screen.getByRole('contentinfo');
    expect(footer).toBeInTheDocument();
    expect(footer).toHaveClass('footer');
  });

  it('displays the copyright info', () => {
    expect(screen.getByText('Â© Dmitrii Suroviagin, 2023')).toBeInTheDocument();
  });

  it('displays 3 links', () => {
    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(3);
    for (const link of links) {
      expect(link).toHaveProperty('target', '_blank');
      expect(link).toHaveProperty('rel', 'noreferrer');
    }
  });

  it('all links have correct href attributes', () => {
    const links = screen.getAllByRole('link');
    expect(links[0]).toHaveProperty('href', externalLinks.mail);
    expect(links[1]).toHaveProperty('href', externalLinks.linkedIn);
    expect(links[2]).toHaveProperty('href', externalLinks.gitHub);
  });

  it('all links have icons', () => {
    const links = screen.getAllByRole('link');
    expect(links[0]).toHaveClass(Icon.Mail);
    expect(links[1]).toHaveClass(Icon.LinkedIn);
    expect(links[2]).toHaveClass(Icon.Github);
  });

  it('all links have texts', () => {
    const links = screen.getAllByRole('link');
    expect(links[0]).toHaveAttribute('title', 'Mail');
    expect(links[1]).toHaveAttribute('title', 'LinkedIn');
    expect(links[2]).toHaveAttribute('title', 'GitHub');
  });

  it('should match the snapshot', () => {
    const { asFragment } = renderWithRedux(<Footer {...testProps} />, mockedReducer, mockedState);
    expect(asFragment()).toMatchSnapshot();
  });
});
