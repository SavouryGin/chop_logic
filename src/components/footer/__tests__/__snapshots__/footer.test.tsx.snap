// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component: should match the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer test-class"
  >
    <span
      class="footer__copyright"
    >
      Â© Dmitrii Suroviagin, 2023
    </span>
    <span
      class="footer__links"
    >
      <a
        class="chop-icon__mail4"
        href="mailto:Dmitrii.Suroviagin@gmail.com"
        rel="noreferrer"
        target="_blank"
        title="Mail"
      />
      <a
        class="chop-icon__linkedin-square"
        href="https://www.linkedin.com/in/dmitrii-suroviagin/"
        rel="noreferrer"
        target="_blank"
        title="LinkedIn"
      />
      <a
        class="chop-icon__github"
        href="https://github.com/SavouryGin"
        rel="noreferrer"
        target="_blank"
        title="GitHub"
      />
    </span>
  </footer>
</DocumentFragment>
`;
